RecoTracker/SpecialSeedGenerators/interface/CosmicTrackingRegion.h

```diff
@@ -83,18 +83,18 @@ class CosmicTrackingRegion : public TrackingRegionBase {
 
   TrackingRegion::Hits hits(const edm::EventSetup& es, const SeedingLayerSetsHits::SeedingLayer& layer) const override;
 
-  HitRZCompatibility* checkRZ(const DetLayer* layer,
-                              const Hit& outerHit,
-                              const edm::EventSetup& iSetup,
-                              const DetLayer* outerlayer = nullptr,
-                              float lr = 0,
-                              float gz = 0,
-                              float dr = 0,
-                              float dz = 0) const override {
+  std::unique_ptr<HitRZCompatibility> checkRZ(const DetLayer* layer,
+                                              const Hit& outerHit,
+                                              const edm::EventSetup& iSetup,
+                                              const DetLayer* outerlayer = nullptr,
+                                              float lr = 0,
+                                              float gz = 0,
+                                              float dr = 0,
+                                              float dz = 0) const override {
     return nullptr;
   }
 
-  CosmicTrackingRegion* clone() const override { return new CosmicTrackingRegion(*this); }
+  std::unique_ptr<TrackingRegion> clone() const override { return std::make_unique<CosmicTrackingRegion>(*this); }
```

Comment:
Are you replacing CosmicTrackingRegion into TrackingRegion on purpose?

Reply:
Yes, covariant return types work only with raw pointers, not with smart pointers.